src/spec-lang/tc/typecheck.ts

```diff
@@ -1648,14 +1648,32 @@ function matchArguments(
     callable: FunctionDefinition | FunctionType
 ) {
     const funT = callable instanceof FunctionDefinition ? getFunDefType(callable) : callable;
+    const isVariadic = funT.parameters.length > 0 && last(funT.parameters) instanceof VariableTypes;
 
-    if (argTs.length !== funT.parameters.length) {
+    // For non-variadic functions the number of arguments must match the number of formal parameters
+    if (!isVariadic && argTs.length !== funT.parameters.length) {
+        return false;
+    }
+
+    // For variadic functions the number of arguments must be at least the
+    // number of formal parameters minus one (the variable types can match 0
+    // types)
+    if (isVariadic && argTs.length < funT.parameters.length - 1) {
         return false;
     }
 
     for (let i = 0; i < funT.parameters.length; i++) {
         const formalT = funT.parameters[i];
 
+        if (formalT instanceof VariableTypes) {
+            assert(
+                i === funT.parameters.length - 1,
+                `Unexpected variable type not in last position of fun {1}`,
```

Comment:
Note that `assert()` placeholders [are zero-based](https://github.com/ConsenSys/solc-typed-ast/blob/ae66129bb5c8e49007ea56c882ff590df38a174d/src/misc/utils.ts#L26-L35). Message should rever `{0}` instead:
```
Unexpected variable type not in last position of fun {0}
```

Reply:
Fixed in 8bb7a79.